packages/event-handler/tests/unit/rest/BaseRouter.test.ts

```diff
@@ -583,6 +1021,133 @@ describe('Class: BaseRouter', () => {
+        @app.get('/test', [middleware])
+        public async getTest() {
+          executionOrder.push('handler');
+          return { result: 'decorator-with-middleware' };
+        }
+
+        public async handler(event: unknown, context: Context) {
+          return app.resolve(event, context);
+        }
+      }
+
+      const lambda = new Lambda();
+
+      // Act
+      const result = await lambda.handler(
+        createTestEvent('/test', 'GET'),
+        context
+      );
+
+      // Assess
+      expect(executionOrder).toEqual([
+        'decorator-middleware-start',
+        'handler',
+        'decorator-middleware-end',
+      ]);
+      expect(result).toEqual({
+        statusCode: 200,
+        body: JSON.stringify({ result: 'decorator-with-middleware' }),
+        headers: { 'Content-Type': 'application/json' },
+        isBase64Encoded: false,
+      });
+    });
+
+    it.each([
+      ['GET', { result: 'get-decorator-middleware' }],
+      ['POST', { result: 'post-decorator-middleware' }],
+      ['PUT', { result: 'put-decorator-middleware' }],
+      ['PATCH', { result: 'patch-decorator-middleware' }],
+      ['DELETE', { result: 'delete-decorator-middleware' }],
+      ['HEAD', { result: 'head-decorator-middleware' }],
+      ['OPTIONS', { result: 'options-decorator-middleware' }],
+    ])(
+      'routes %s requests with decorator middleware',
+      async (method, expected) => {
+        // Prepare
+        const app = new TestResolver();
+        const executionOrder: string[] = [];
+        const middleware = createTrackingMiddleware(
+          `${method.toLowerCase()}-middleware`,
+          executionOrder
+        );
+
+        class Lambda {
+          @app.get('/test', [middleware])
+          public async getTest() {
+            return { result: 'get-decorator-middleware' };
+          }
+
+          @app.post('/test', [middleware])
+          public async postTest() {
+            return { result: 'post-decorator-middleware' };
+          }
+
+          @app.put('/test', [middleware])
+          public async putTest() {
+            return { result: 'put-decorator-middleware' };
+          }
+
+          @app.patch('/test', [middleware])
+          public async patchTest() {
+            return { result: 'patch-decorator-middleware' };
+          }
+
+          @app.delete('/test', [middleware])
+          public async deleteTest() {
+            return { result: 'delete-decorator-middleware' };
+          }
+
+          @app.head('/test', [middleware])
+          public async headTest() {
+            return { result: 'head-decorator-middleware' };
+          }
+
+          @app.options('/test', [middleware])
+          public async optionsTest() {
+            return { result: 'options-decorator-middleware' };
+          }
+
+          public async handler(event: unknown, context: Context) {
+            return app.resolve(event, context);
+          }
+        }
+
+        const lambda = new Lambda();
```

Comment:
Should we change one of these tests to use the `const handler = lambda.handler.bind(lambda)` and add the `{ scope: this }` in the resolve call, so we test the scope preservation?

As far as I can see we're not doing it in this test, and I see we've modified decorator code - but if I missed it or it's covered elsewhere you can ignore, although I try to write it like that always so we spot regressions in multiple places.

Reply:
There are tests for scope preservation eleswhere in suite, we don't preserve `this` in middleware so I didn't added but it does no harm to have it here.